client/src/components/pages/personalProfile/index.js

```diff
@@ -0,0 +1,187 @@
+import React from 'react'
+import { Form, Input } from 'antd'
+import Button from '../../utils/Button'
+
+import 'antd/dist/antd.css'
+import './style.css'
+
+class PersonalForm extends React.Component {
+
+  state = {
+    confirmDirty: false,
+  };
+
+  handleSubmit = e => {
+    e.preventDefault();
+    this.props.form.validateFieldsAndScroll((err, values) => {
+      if (!err) {
+        console.log('Received values of form: ', values);
+      }
+    });
+  };
+
+  handleConfirmBlur = e => {
+    const value = e.target.value;
+    this.setState({ confirmDirty: this.state.confirmDirty || !!value });
+  };
+
+  compareToFirstPassword = (rule, value, callback) => {
+    const form = this.props.form;
+    if (value && value !== form.getFieldValue('password')) {
+      callback('Password must match!');
+    } else {
+      callback();
+    }
+  };
+
+  validateToNextPassword = (rule, value, callback) => {
+    const form = this.props.form;
+    if (value && this.state.confirmDirty) {
```

Comment:
@AbdallahAmmar96 I wonder if it would be simpler to just call validateFields if `value && !!form.getFieldValue('confirm')`, instead of updating and using the confirmDirty flag.